tests/test_flash_attn_ck.py

```diff
@@ -0,0 +1,215 @@
+import math
+
+import pytest
+import torch
+import torch.nn.functional as F
+from einops import rearrange, repeat
+from flash_attn import (
+    flash_attn_func,
+    flash_attn_qkvpacked_func,
+    flash_attn_varlen_qkvpacked_func,
+)
+
+from test_flash_attn import (
+    attn_bias_from_alibi_slopes,
+    convert_flash_attn_S_to_softmax,
+    generate_qkv,
+    attention_qkvpacked_ref,
+    generate_random_padding_mask
+)
+
+is_gfx94x = torch.cuda.get_device_capability("cuda") == (9, 4)
+
+@pytest.mark.parametrize("dtype", [torch.float16, torch.bfloat16])
+@pytest.mark.parametrize("deterministic", [False])
+@pytest.mark.parametrize("alibi", [False, True])
+@pytest.mark.parametrize("local", [False, True])
+@pytest.mark.parametrize("causal", [False, True])
+@pytest.mark.parametrize("d", [32, 40, 59, 64, 80, 96, 111, 128, 160, 192, 224, 256])
+@pytest.mark.parametrize("seqlen", [97, 128, 200, 384, 768, 1024, 1025, 2048])
+@pytest.mark.parametrize("dropout_p", [0.0, 0.17])
+def test_flash_attn_qkvpacked(seqlen, d, dropout_p, causal, local, alibi, deterministic, dtype):
+    if d > 256:
+        pytest.skip()
+
+    device = "cuda"
+    # set seed
+    torch.random.manual_seed(0)
+    batch_size = 4
+    nheads = 9
+    window_size = (-1, -1) if not local else torch.randint(0, seqlen, (2,))
+    #Causal is the special case where window_size_right == 0 and window_size_left < 0.
+    window_size = (-1, 0) if causal else window_size
```

Comment:
could you wrap the logic above into lambda function?

Reply:
Remove window_size = (-1, 0) if causal else window_size, c++ api will handle this.
Sync with test_flash_attn.py
5728fb049233abac8e20b756fd0991d7b21ee330